89.md

```diff
@@ -1,133 +1,232 @@
 NIP-89
 ======
 
-Recommended Application Handlers
---------------------------------
+Application Handlers
+--------------------
 
 `draft` `optional`
 
-This NIP describes `kind:31989` and `kind:31990`: a way to discover applications that can handle unknown event-kinds.
+This NIP describes a way to discover different types of applications that can handle unknown event-kinds.
 
 ## Rationale
 
 Nostr's discoverability and transparent event interaction is one of its most interesting/novel mechanics.
 This NIP provides a simple way for clients to discover applications that handle events of a specific kind to ensure smooth cross-client and cross-kind interactions.
 
-### Parties involved
+## Overview
 
-There are three actors to this workflow:
+This NIP defines "application handlers", which are events intended to advertise support for a certain event kind. These handlers may be for several different types of nostr-related applications, including:
 
-* application that handles a specific event kind (note that an application doesn't necessarily need to be a distinct entity and it could just be the same pubkey as user A)
-    * Publishes `kind:31990`, detailing how apps should redirect to it
-* user A, who recommends an app that handles a specific event kind
-    * Publishes `kind:31989`
-* user B, who seeks a recommendation for an app that handles a specific event kind
-    * Queries for `kind:31989` and, based on results, queries for `kind:31990`
+- `11991` for nostr clients
+- `11992` for nostr DVMs
+- `11993` for nostr relays
 
-## Events
+Applications SHOULD have their own nostr identity, and SHOULD publish a kind 0 profile, a kind 10002 relay list, and any other metadata events that may be helpful for consumers of the applications.
+
+This NIP also defines a `kind:31989` application selection event which allows users to configure their default application handlers.
+
```

Comment:
"... which allows users to select their preferred application handlers."

Since you can configure more than one handler per kind, "default" isn't right.

Reply:
Yeah, I'm not sure what to do with this. Multiple defaults could be ok though, clients would just need a heuristic for selecting one. They could just grab the first one, or they could compare the handler metadata to find the best option (based on platform for example).